table/block_based/block_based_table_builder.cc

```diff
@@ -461,6 +465,12 @@ struct BlockBasedTableBuilder::Rep {
       buffer_limit = std::min(tbo.target_file_size,
                               compression_opts.max_dict_buffer_bytes);
     }
+    // TODO figure out c++ member initialization syntax
+    buffer_exceeds_global_memory_limit = false;
+    cache_rep = (table_options_.no_block_cache
+                     ? nullptr)
+                     : new CacheRep(table_options.block_cache.get());
```

Comment:
Question 1:
While the block_cache in the case of WriteBufferManager is passed in on WBM's creation, in this case, I believe we should account memory in the block cache that is set in the table_options by the block_based_table_factory. Did I understand it correctly?